api/provisioner/provisioner.go

```diff
@@ -57,6 +57,31 @@ func (st *State) Machine(tag names.MachineTag) (*Machine, error) {
 	}, nil
 }
 
+// Machines provides access to methods of a state.Machine through the facade
+// for the given tags.
+func (st *State) Machines(tags []names.MachineTag) ([]*Machine, []params.LifeResult, error) {
```

Comment:
params.LifeResult is used as merely a container for an Error, and the caller in provisioner_task names the return result machineErrors. It would be better to define a struct in this provisioner.go file which would contain the Machine and an error and return a slice of this new struct.  Something like

func (st *State) Machines(tags []names.MachineTag) ([]MachineResult, error)

type MachineResult struct (
    machine *Machine
    err error
}

Then, the processing for task.machineGetter.Machines(machineTags) becomes more straightforward.

Reply:
updated for Machines() and MachinesWithTransientErrors()